### DISCUSSION TITLE
DLL load fails

### DISCUSSION THREAD

**Discussion by caepman** [banesullivan/localtileserver#188] [Q&A] [answered]
Hi:

I've been struggling to get localtileserver working through a conda 3.9 python environment on Windows 10 in a jupyter notebook. As soon as I try to import localtileserver, I get DLL import errors:

![DLL_ImportErrors](https://github.com/banesullivan/localtileserver/assets/79587528/c26e4dc6-134a-4eff-bd87-b411fd12e88d)

I'm not certain how to fix this problem, so any advice will be appreciated!

-Darren

**Comment by giswqs** [comment]
How did you install rasterio? Try creating a fresh conda env to install rasterio. 

```bash
conda create -n geo python=3.9
conda activate geo
conda install -c conda-forge mamba
mamba install -c conda-forge rasterio localtileserver
```

**Comment by caepman** [comment]
Thanks for your reply! I created the virtual python 3.9 environment through conda. To install localtileserver (and dependencies) I also used conda:
`conda install -c conda-forge localtileserver ipyleaflet`
This was following the instruction provided on the [localtileserver site](https://localtileserver.banesullivan.com/installation/index.html).
I believe rasterio and osgeo/gdal were installed as dependencies.

**Reply by banesullivan** [reply]
There may be some issues in the dependencies where gdal and rasterio are conflicting. This will be resolved in the 0.10 release (pre release wheels are in pypi) in the coming weeks.


In the meantime, perhaps making sure GDAL isn't installed will help?

**Reply by caepman** [reply]
I'll try uninstalling GDAL to see if that can resolve the error I've encountered. I forgot to mention in my reply to giswqs that I believe I am currently on the sixth or seventh virtual environment I've created to try to get localtileserver up an running. I've tried various combinations of installing packages to get things up an running, without success thus far. Advice provided thus far is much apppreciated!

**Reply by giswqs** [reply]
Try creating a new conda env and install GAL with the following comand:

```bash
pip install -U --find-links https://girder.github.io/large_image_wheels GDAL
```

Test if GDAL has been installed successfully:

```python
from osgeo import gdal
```

If GDAL can be imported successfuly, then proceed to instlal localtileserver. Otherwise, it is a GDAL installation issue.

**Reply by caepman** [reply]
An update: Following advice from giswqs I was able to get GDAL installed and imported from the python prompt without errors. However, when moving to the next step. installing localtileserver and rasterio... dll errors were generated. I read in the docs for rasterio about issues it can have with GDAL, specifically, ["mutual incompatibilities"](https://rasterio.readthedocs.io/en/stable/topics/switch.html#the-gdal-environment). I'm wondering if this is related to what banesullivan suggested above and if this is what is causing problems for me, working through Windows 10, 64-bit. I'm currently creating a new Python 3.9 venv and will move forward trying to install localtileserver v010.0rc2 to see how far I can get with it.

**Reply by banesullivan** [reply]
Let me know if the latest 0.10 release candidate works for you... i'm hoping it does 🤞🏻

**Reply by caepman** [reply, marked as answer]
Great news... I was able to get the localtileserver v010.0rc2 package installed and working in a Python 3.9 venv running on my Windows 10, 64-bit computer!

I had some trouble when I first tried to install v010.0rc2... I was at the office, behind our firewall. Once I was home and off our network the install worked without a problem. I used the localtileserver-0.10.0rc2-py3-none-any.whl file I downloaded from your GitHub (I had downloaded the wheel while at work, after the regular pip install failed).

After confirming I had localtileserver working using the San Francisco example, I tried one of my own small *.tif files, and it worked. Next, I loaded a near 2 GB tif... I'm very impressed with the loading, zooming, and panning! 

LocalTileServer will be extremely helpful for the work I have planned - thanks for creating this package and for the advice you provided to help me get it installed!
